modules/swagger-codegen/src/main/resources/Java/ApiClient.mustache

```diff
@@ -651,4 +660,23 @@ public class ApiClient {
     }
     return hostMap.get(basePath);
   }
+  
+  /**
+   * Adds pattern-based parameters (after name validation) to a dictionary query parameters.
+   * @throws ApiException 
+   */
+  public <T> void addPatternQueryParameters(String collectionFormat, Map<String, T> parameters, String pattern, List<Pair> queryParams) throws ApiException 
+  {
+    Pattern p = Pattern.compile(pattern);
+    for (Entry<String, T> entry: parameters.entrySet())
+    {
+      if (p.matcher(entry.getKey()).matches())
+      {
+        queryParams.addAll(parameterToPairs(collectionFormat, entry.getKey(), entry.getValue()));
```

Comment:
why not return the List<Pair> and do the queryParams.addAll outside of the function? It is not so clear that the queryParams is the "out" parameter. But I leave it to you. Just a suggestion...